src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/documents/TextBoxView.cs

```diff
@@ -44,6 +44,8 @@ internal TextBoxView(ITextBoxViewHost host)
         {
             Invariant.Assert(host is Control);
             _host = host;
+
+            ((Control)_host).Unloaded += OnHostUnloaded;
```

Comment:
I think this code path can be reached multiple times in the lifetime of a TextBox (via AttachFromVisualTree/DetachFromVisualTree pairs) so you need to unsubscribe when TextBoxView is detached.

Generally I'd prefer if the owner would notify the TextBoxView instead of the TextBoxView subscribing with the owner.

Reply:
Will have a look and change it to the way you suggested. Just give me a few days.

Reply:
The best way to achieve this would be to add the event subscription code to `EnsureTextContainerListeners` and to `RemoveTextContainerListeners` as those are already called from the host to ensure proper initialization and disposal.
What do you think about that?

Reply:
Yeah that should work and is probably intended for that purpose, didn't see it yesterday when I did a quick check of who calls the constructor.